cmake/OpenCVDetectOpenCL.cmake

```diff
@@ -2,11 +2,26 @@ set(OPENCL_FOUND ON CACHE BOOL "OpenCL library is found")
 if(APPLE)
   set(OPENCL_LIBRARY "-framework OpenCL" CACHE STRING "OpenCL library")
   set(OPENCL_INCLUDE_DIR "" CACHE PATH "OpenCL include directory")
-else(APPLE)
+else()
   set(OPENCL_LIBRARY "" CACHE STRING "OpenCL library")
-  set(OPENCL_INCLUDE_DIR "${OpenCV_SOURCE_DIR}/3rdparty/include/opencl/1.2" CACHE PATH "OpenCL include directory")
+  if(WITH_OPENCL_D3D11_NV)
+    unset(OPENCL_INCLUDE_DIR CACHE)
```

Comment:
Unsetting of cached variables is a bad practice. These variables are dedicated for configuration from "User".

You should check for `cl_d3d11_ext.h` in specified directory `OPENCL_INCLUDE_DIR`.
- found: enable feature: `set(HAVE_OPENCL_D3D11_NV ON)`
- not found: print warning, keep feature disabled.

Reply:
You're right.And if detection fails, it can also be searched from environment variables.

Reply:
Done